chartfx-chart/src/test/java/de/gsi/chart/ui/css/CssPropertyFactoryTest.java

```diff
@@ -0,0 +1,161 @@
+        verifyThat(styleable.getTestEnumP(), equalTo(TestEnum.DISABLED));
+        verifyThat(styleable.getPseudoClassStates(), Matchers.contains(PseudoClass.getPseudoClass("disabled")));
+        verifyThat(styleable.getPseudoClassStates(), not(contains(PseudoClass.getPseudoClass("enabled"))));
+        fxRobot.interact(() -> styleable.setStyle("-fx-test-double: 2.3; -fx-test-int: 23; -fx-test-string: styletest;-fx-test-bool: true;-fx-test-object: green;-fx-test-enum: disabled;-fx-test-enum-p: enabled; -fx-min-height: 13;"));
+        verifyThat(styleable.getMinHeight(), equalTo(13.0)); // parent property
+        verifyThat(styleable.getTestDouble(), equalTo(2.3));
+        verifyThat(styleable.getTestObject(), equalTo(Color.GREEN));
+        verifyThat(styleable.getTestInt(), equalTo(23));
+        verifyThat(styleable.getTestString(), equalTo("styletest"));
+        verifyThat(styleable.getTestEnum(), equalTo(TestEnum.DISABLED));
+        verifyThat(styleable.getTestEnumP(), equalTo(TestEnum.ENABLED));
+        verifyThat(styleable.getTestBool(), equalTo(true));
+        verifyThat(styleable.getPseudoClassStates(), not(contains(PseudoClass.getPseudoClass("disabled"))));
+        verifyThat(styleable.getPseudoClassStates(), contains(PseudoClass.getPseudoClass("enabled")));
+
+        // test applying via css sheet
+        // test with multiple instances
+    }
+
+    public static class TestStyleable extends Region {
+        private static final CssPropertyFactory<TestStyleable> CSS = new CssPropertyFactory<>(Region.getClassCssMetaData());
+
+        @Override
+        public List<CssMetaData<? extends Styleable, ?>> getCssMetaData() {
+            return CSS.getCssMetaData();
+        }
+
+        // double
+        private final StyleableDoubleProperty testDouble = CSS.createDoubleProperty(this, "testDouble", "-fx-test-double", s -> s.testDouble, 4.2, true, null);
+
+        public double getTestDouble() {
+            return testDouble.getValue();
+        }
+
+        // float
+        private final StyleableFloatProperty testFloat = CSS.createFloatProperty(this, "testFloat", "-fx-test-float", s -> s.testFloat, 1.337f, true, null);
+
+        public float getTestFloat() {
+            return testFloat.getValue();
+        }
+
+        // int
+        private final StyleableIntegerProperty testInt = CSS.createIntegerProperty(this, "testInt", "-fx-test-int", s -> s.testInt, 42, true, null);
+
+        public int getTestInt() {
+            return testInt.getValue();
+        }
+
+        // long
+        private final StyleableLongProperty testLong = CSS.createLongProperty(this, "testLong", "-fx-test-long", s -> s.testLong, 123456789123456789l, true, null);
+
+        public long getTestLong() {
+            return testLong.getValue();
+        }
+
+        // string
+        private final StyleableStringProperty testString = CSS.createStringProperty(this, "testString", "-fx-test-string", s -> s.testString, "test", true, null);
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5741094)